src/libraries/System.IO.Hashing/src/System/IO/Hashing/Crc32.cs

```diff
@@ -169,6 +169,16 @@ public static int Hash(ReadOnlySpan<byte> source, Span<byte> destination)
 
         private static uint Update(uint crc, ReadOnlySpan<byte> source)
         {
+#if NET7_0_OR_GREATER
+            if (System.Runtime.Intrinsics.Vector128.IsHardwareAccelerated
+                && System.Runtime.Intrinsics.X86.Sse2.IsSupported
+                && System.Runtime.Intrinsics.X86.Pclmulqdq.IsSupported)
+            {
+                crc = UpdateX86(crc, ref source);
```

Comment:
This will make `source` address-taken that is likely to regress performance of the loop below.

Reply:
I've rewritten this to avoid passing source by reference